Sources/tart/Commands/List.swift

```diff
@@ -5,18 +5,26 @@ import SwiftUI
 struct List: AsyncParsableCommand {
   static var configuration = CommandConfiguration(abstract: "List created VMs")
 
-  @Flag(name: [.short, .long], help: ArgumentHelp("Only display VM names"))
+  @Flag(name: [.short, .long], help: ArgumentHelp("Only display VM names."))
   var quiet: Bool = false
 
+  @Option(name: [.short, .long], help: ArgumentHelp("Only display VMs of this Source."))
+  var source: String = ""
+
   func run() async throws {
     do {
-      if !quiet {
-        print("Source\tName")
+      switch source {
+      case "local":
+        displayTable("local", try VMStorageLocal().list())
+      case "oci":
+        displayTable("oci", try VMStorageOCI().list().map { (name, vmDir, _) in (name, vmDir) })
+      case "":
+        displayTable("local", try VMStorageLocal().list())
+        displayTable("oci", try VMStorageOCI().list().map { (name, vmDir, _) in (name, vmDir) })
```

Comment:
I think here we will print the header twice, because it is printed on each `displayTable()` call now.

You can probably move the validation logic into `validate()`, like in `tart delete`:

https://github.com/cirruslabs/tart/blob/7a2c20ba30e8e2e25a5b2ad035e2685cb9b54954/Sources/tart/Commands/Create.swift#L21-L25

...and revert the header change.

Reply:
@edigaryev if you think it makes more sense I could introduce an [EnumerableFlag](https://apple.github.io/swift-argument-parser/documentation/argumentparser/enumerableflag) so that the sources are validated automatically by ArgumentParser and mutual exclusivity is enforced.

Users would run `tart list --local` or `tart list --oci`. It depends what we have in mind in terms of sources for the future.

Reply:
I don't really like this approach as it needlessly pollutes the flags namespace.

You have to manually search for all the possible variants (or name them like `--source-local`, `--source-oci`, etc., which makes little sense if we can just have `--source <source>`), instead of looking for a documentation for a single option, and it would be pain to deprecate them in the future.

I'm actually leaning more leaning towards Docker's approach, where there's only one `--filter` argument (see `docker image ls --help`, for example), but your current approach is sensible too, probably a good middle ground between `--filter=...` and `--local`/`--oci`.

Reply:
Good points. 
I'll fix and move forward with my initial approach.